crates/rustc_codegen_spirv/src/codegen_cx/type_.rs

```diff
@@ -156,7 +156,6 @@ impl<'tcx> BaseTypeMethods<'tcx> for CodegenCx<'tcx> {
     fn type_kind(&self, ty: Self::Type) -> TypeKind {
         match self.lookup_type(ty) {
             SpirvType::Void => TypeKind::Void,
-            SpirvType::Bool => TypeKind::Integer, // thanks llvm
             SpirvType::Integer(_, _) => TypeKind::Integer,
```

Comment:
This looks like a lint bug, it should suggest collapsing these two cases, but somehow it finds the other ones down below instead?

Reply:
It did suggest collapsing them, that's why I had to move it down to the integer arm, but formatted it a bit differently to keep the comment.

Would indeed be good if the lint took comments into consideration, as we talked about on discord. but this is the best I could get it to do without that.

any ideas on how to improve it here in the code?